resources/measures/ResidentialDemandResponse/measure.rb

```diff
@@ -0,0 +1,369 @@
+    dr_schedule_heat.setDescription("File name of the csv that contains hourly DR signals of -1, 0, or 1 for the heating setpoint schedule.")
+    dr_schedule_heat.setDefaultValue("none")
+    args << dr_schedule_heat	
+	
+	# MAke a string argument for offset magnitude for temperature setpoint DR events
+    offset_magnitude_heat = OpenStudio::Measure::OSArgument::makeDoubleArgument("offset_magnitude_heat", true)
+    offset_magnitude_heat.setDisplayName("Heating DR Offset Magnitude")
+    offset_magnitude_heat.setDescription("The magnitude of the heating setpoint offset, which is applied to non-zero hours specified in the DR schedule. The offset should be positive")
+    offset_magnitude_heat.setUnits("degrees F")
+    offset_magnitude_heat.setDefaultValue(0)
+    args << offset_magnitude_heat
+    
+	# make an argument for 8760 DR schedule csv file
+	dr_schedule_cool = OpenStudio::Measure::OSArgument::makeStringArgument("dr_schedule_cool", true)
+    dr_schedule_cool.setDisplayName("Cooling Setpoint DR Schedule File Name")
+    dr_schedule_cool.setDescription("File name of the csv that contains hourly DR signals of -1, 0, or 1 for the cooling setpoint schedule.")
+    dr_schedule_cool.setDefaultValue("none")
+    args << dr_schedule_cool	
+	
+	# MAke a string argument for offset magnitude for temperature setpoint DR events
+    offset_magnitude_cool = OpenStudio::Measure::OSArgument::makeDoubleArgument("offset_magnitude_cool", true)
+    offset_magnitude_cool.setDisplayName("Cooling DR Offset Magnitude")
+    offset_magnitude_cool.setDescription("The magnitude of the heating setpoint offset, which is applied to non-zero hours specified in the DR schedule. The offset should be positive")
+    offset_magnitude_cool.setUnits("degrees F")
+    offset_magnitude_cool.setDefaultValue(0)
+    args << offset_magnitude_cool	
+	return args
+  end
+  
+  def run(model, runner, user_arguments)
+    super(model, runner, user_arguments)
+    
+	if !runner.validateUserArguments(arguments(model), user_arguments)
+      return false
+    end    
+    
+	weather = WeatherProcess.new(model, runner)
+    if weather.error?
+      return false
+    end
+	
+	# Import data and create DR schedule
+    dr_dir = runner.getStringArgumentValue("dr_directory", user_arguments)
+	offset_heat = runner.getDoubleArgumentValue("offset_magnitude_heat", user_arguments)
+	dr_sch_htg = runner.getStringArgumentValue("dr_schedule_heat", user_arguments)
+    offset_cool = runner.getDoubleArgumentValue("offset_magnitude_cool", user_arguments)
+	dr_sch_clg = runner.getStringArgumentValue("dr_schedule_cool", user_arguments)	
+
+	# Finished Zones
+	finished_zones = []
+    model.getThermalZones.each do |thermal_zone|
+      if Geometry.zone_is_finished(thermal_zone)
+        finished_zones << thermal_zone
+      end
+    end
+    
+	# Check for setpoint offset
+	if offset_heat==0 and offset_cool==0
+	  runner.registerInfo("DR offset magnitude set to 0")
```

Comment:
I'd be more explicit here about what's happening. Like "User specified a DR offset magnitude of zero; skipping measure." Or something.